contrib/Intel/JenkinsBuildScripts/tests.py

```diff
@@ -0,0 +1,311 @@
+import sys
+import os
+
+print(os.environ['CI_SITE_CONFIG'])
+sys.path.append(os.environ['CI_SITE_CONFIG']) # for adding path for cisite_config
+
+import subprocess
+import re
+import cisite_config
+import common
+import shlex
+
+class test:
+    def __init__ (self, branchname, buildno,  testname, core_prov, fabric, hosts, util_prov=None):
+        self.branchname = branchname
+        self.buildno = buildno
+        self.testname = testname
+        self.core_prov = core_prov
+        self.util_prov = "ofi_{}".format(util_prov) if util_prov != None else "" 
+        self.fabric = fabric
+        self.hosts = hosts
+        if (len(hosts) == 2):
+            self.server = hosts[0]
+            self.client = hosts[1]
+       
+        self.nw_interface = cisite_config.interface_map[self.fabric] # include siteconfig
+        self.libfab_installpath = "{}/{}/{}".format(cisite_config.install_dir, self.branchname, self.buildno) 
+        self.env = [("FI_VERBS_MR_CACHE_ENABLE", "1"),("FI_VERBS_INLINE_SIZE", "256")] if self.core_prov == "verbs" else []
+                     
+
+    def get_exclude_file(self):
+        path = self.libfab_installpath
+        efile_path = "{}/share/fabtests/test_configs".format(path)
+
+        prov = self.util_prov if self.util_prov else self.core_prov
+        efile_old = "{path}/{prov}/{prov}.exclude".format(path=efile_path, prov=prov)
+        if self.util_prov:
+            efile = "{path}/{util_prov}/{core_prov}/exclude".format(
+            path=efile_path, util_prov=self.util_prov, core_prov=self.core_prov)
+        else:
+            efile = "{path}/{prov}/exclude".format(path=efile_path, prov=self.core_prov)
+           
+        if os.path.isfile(efile):
+            return efile
+        elif os.path.isfile(efile_old):
+            return efile_old
+        else:
+            print("Exclude file: {} not found!".format(efile))
+            return None  
+
+class fabtest(test):
+    
+    def __init__(self, branchname, buildno, testname, core_prov, fabric, hosts, util_prov=None):
+        super().__init__(branchname, buildno, testname, core_prov, fabric, hosts, util_prov)
```

Comment:
You may be able to use *args instead of listing all the non-keyword args individually.